src/algebra/quaternion.lean

```diff
@@ -573,3 +585,41 @@ monoid_with_zero_hom.map_div norm_sq a b
 end field
 
 end quaternion
+
+namespace cardinal
+
+open_locale cardinal quaternion
+
+/-- A useful lemma for the following theorems. -/
+lemma pow_four {A : Type*} : #A * #A * #A * #A = #A ^ 4 :=
+by ring
+
+section quaternion_algebra
+
+variables {R : Type*} (c₁ c₂ : R)
+
+/-- The cardinality of a quaternion algebra, as a type. -/
+@[simp] lemma mk_quaternion_algebra : #ℍ[R, c₁, c₂] = #R ^ 4 :=
+by { rw mk_congr (quaternion_algebra.equiv_prod c₁ c₂), simp only [mk_prod, lift_id], ring }
+
+/-- The cardinality of a quaternion algebra, as a set. -/
+@[simp] lemma mk_univ_quaternion_algebra : #(set.univ : set ℍ[R, c₁, c₂]) = #R ^ 4 :=
+by rw [mk_univ, mk_quaternion_algebra]
+
+end quaternion_algebra
+
+section quaternion
+
+variables (R : Type*) [has_one R] [has_neg R]
+
+/-- The cardinality of the quaternions, as a type. -/
+@[simp] lemma mk_quaternion : #ℍ[R] = #R ^ 4 :=
+mk_quaternion_algebra _ _
+
+/-- The cardinality of the quaternions, as a set. -/
+@[simp] lemma mk_univ_quaternion : #(set.univ : set ℍ[R]) = #R ^ 4 :=
+mk_univ_quaternion_algebra _ _
+
```

Comment:
I guess that if `R` is infinite, then `#R ^4 = #R`, right? That's the most common case for quaternions. People typically consider quaternion algebras over `ℚ` or `ℝ`.
Could you please add a bit of glue for that case?